tests/python/unittest/test_tir_transform_bf16_legalize.py

```diff
@@ -0,0 +1,152 @@
+                            to32(a[i]),
+                            to32(b[i]),
+                        )
+                    )
+                )
+            )
+        ))
+        s = te.create_schedule(c.op)
+        stmt = lower_stmt(s, [a, b, c], tvm.tir.transform.BF16CastElimination)
+        return stmt
+
+    def get_target():
+        a = te.placeholder((100,), dtype='bfloat16')
+        b = te.placeholder((100,), dtype='bfloat16')
+        c = te.compute((100,), lambda i: to16(
+            topi.add(topi.add(
+                        to32(a[i]),
+                        to32(b[i]),
+                    ),
+                    topi.add(
+                        to32(a[i]),
+                        to32(b[i]),
+                    )
+                )
+        ))
+        s = te.create_schedule(c.op)
+        func = tvm.driver.build_module.form_irmodule(s, [a,b,c], "main", None)["main"]
+        return func.body
+    tvm.ir.assert_structural_equal(get_eliminated(), get_target())
+
+def test_legalize():
+    def to32(v):
+        uint32_v = topi.cast(v, "uint32")
+        uint32_v = tvm.tir.call_pure_intrin("uint32", "shift_left", uint32_v, tvm.tir.const(16, "uint32"))
+        return tvm.tir.call_pure_intrin("float32", "reinterpret", uint32_v)
+    def to16(v):
+        uint32_v = tvm.tir.call_pure_intrin("uint32", "reinterpret", v)
+        rounding_bias = tvm.tir.call_pure_intrin("uint32", "shift_right", uint32_v, tvm.tir.const(16, "uint32"))
+        rounding_bias = tvm.tir.call_pure_intrin("uint32", "bitwise_and", rounding_bias, tvm.tir.const(1, "uint32"))
+        rounding_bias = rounding_bias + tvm.tir.const(0x7FFF, "uint16")
+        uint32_v = uint32_v + rounding_bias
+        uint32_v = tvm.tir.call_pure_intrin("uint32", "shift_right", uint32_v, tvm.tir.const(16, "uint32"))
+        return topi.cast(uint32_v, 'uint16')
+
+    def check(fcompute_before, fcompute_after):
+        a = te.placeholder((100,), dtype='bfloat16', name = 'A')
+        b = te.placeholder((100,), dtype='bfloat16', name = 'B')
+        c = te.compute((100,), fcompute_before(a,b), name = 'C')
+        s = te.create_schedule(c.op)
+        stmt = lower_stmt(s, [a, b, c], tvm.tir.transform.BF16Legalize)
+
+        a = te.placeholder((100,), dtype='uint16', name = 'A')
+        b = te.placeholder((100,), dtype='uint16', name = 'B')
+        c = te.compute((100,), fcompute_after(a,b), name = 'C')
+        s = te.create_schedule(c.op)
+        func = tvm.driver.build_module.form_irmodule(s, [a,b,c], "main", None)["main"]
+        tvm.ir.assert_structural_equal(stmt, func.body)
+
+    def orig1(a,b):
+        return lambda i: a[i]+b[i]+a[99-i]+b[99-i]
+    def after1(a,b):
+        return lambda i: to16(to32(a[i])+to32(b[i])+to32(a[99-i])+to32(b[99-i]))
+    def orig2(a,b):
+        return lambda i: a[i]*b[i]+a[99-i]*b[99-i]+a[i]
+    def after2(a,b):
+        return lambda i: to16(to32(a[i])*to32(b[i])+to32(a[99-i])*to32(b[99-i])+to32(a[i]))
```

Comment:
i am not so why the coding style here can pass pylint...Mind sending a simple fix?

Reply:
Sorry for that. Now I have formatted and manually run pylint on this test file. BTW, the test python files are never checked in TVM CI's pylint :)

Reply:
Ooops I see. That makes sense then :-)